scene/gui/spin_box.cpp

```diff
@@ -114,8 +121,13 @@ void SpinBox::_line_edit_input(const Ref<InputEvent> &p_event) {
 void SpinBox::_range_click_timeout() {
 	if (!drag.enabled && Input::get_singleton()->is_mouse_button_pressed(MouseButton::LEFT)) {
 		bool up = get_local_mouse_position().y < (get_size().height / 2);
-		double step = get_custom_arrow_step() != 0.0 ? get_custom_arrow_step() : get_step();
-		set_value(get_value() + (up ? step : -step));
+		double step = get_step();
+		//arrow button is being pressed, so we also need to set the step to the same value as custom_arrow_step if its not 0
```

Comment:
```suggestion
		// Arrow button is being pressed, so we also need to set the step to the same value as custom_arrow_step if its not 0.
```
Same in the other comment.